lib/Doctrine/DBAL/Driver/OCI8/OCI8Connection.php

```diff
@@ -151,7 +151,11 @@ public function lastInsertId($name = null)
             return false;
         }
 
-        OraclePlatform::assertValidIdentifier($name);
+        $identifiers = explode('.', $name);
+
+        foreach ($identifiers as $identifier) {
```

Comment:
True that but that will fail on DB level then. Not to be validated here IMO. Don't think the PDO equivalent will do something like that either.